### ISSUE TITLE
Smarttube can not be installed on Android TV 10

### ISSUE THREAD

**Issue by chfriedrich** [yuliskov/SmartTube#3393]
Hi,
I tried to install the most recent stable (28544) as well as beta (79015) release on my Sony Bravia VH1 with Android 10 via FX explorer. FX explorer has permissions to install from unsafe sources.
Anyway, the app could not be installed because Google Play Protect says "Unsafe App blocked" with the hint "This app  was built for an older version of Android and does not include the latest privacy protections.

![IMG_0518](https://github.com/yuliskov/SmartTube/assets/73816409/1b3a2d46-3dd1-4df9-a762-bb89721918d9)
![IMG_0517](https://github.com/yuliskov/SmartTube/assets/73816409/66e07ca6-3016-4f75-9bb1-8f621a471c41)

In fact, my Android TV version 10 is quite old...
Is this a known problem? How do I fix it?

**Comment by Lubje** [comment]
Open the Play Store App.
Click on your avatar in the top right corner.
Go to Play Protect. 
Disable "Scan apps with Play Protect".
Install SmartTube.

**Comment by chfriedrich** [comment]
Thanks for the help!
Unfortunatelly on Android 10 there is no option to disable Play Protect, or at least not at my Sony TV...
Sony does not provide an update to a newer android version, so I think I can not use Smarttube.

**Comment by chfriedrich** [comment]
Do you think it is possible to compile smarttube for Android TV 10?
Could you give me a hint which toolchain I should use for that? I never compiled an android app before...

**Comment by EmilyBonar** [comment]
On my Android 12 Google TV the option to disable Play Protect is under Settings > Apps > Security, then turn off "Scan apps with Play Protect".

**Comment by Haris1977** [comment]
@chfriedrich did you try to download the FTVNews downloader and enter the kutt.it/stn_stable command? Does this work?

**Comment by Haris1977** [comment]
I can confirm that i installed smarttubenext on mySony Bravia android 11 tv without any issues. As @Lubje commented i had to disable the "scan apps with Play Protect" otherwise it wont install. 

I dont know what happens on Android 10 though. 

One small question if anyone knows: will the app update normally (when next update will come) if i have the "scan apps with Play Protect" enabled? Or do i need to disable it in order to update?

**Comment by chfriedrich** [comment]
I know that it should be possible to disable play protect in Settings->Apps->Security&Restrictions on Android TV 10. But on my tv, there is only one entry in Security&Restrictions, which is "Unknown sources". There I can give single apps the permission to install apps from unknown sources. I have already allowed this for all apps.

Anyway, I found a workaround: Under Settings->Apps->Google Play Store, I can "Force quit" and "Disable" the play store. I did this and installed Smarttube. No play protect scan was made, the installation was successful. Then I enabled the play store again, and Smarttube is still working.

**Comment by habl** [comment]
Had the same problem on my MiBox 4. For me the fix was similar but on a different location: Settings > Device Preferences > Security & restrictions > Turn off Verify apps.

**Comment by rsauve** [comment]
> > However, I found a workaround: Under Settings->Apps->Google Play Store, I can "Force quit" and "Disable" the play store. I did this and installed Smarttube. No play protect scan was made, the installation was successful. Then I enabled the play store again, and Smarttube is still working.
> 

Yes, I can confirm that this workaround works. I have a TV: **_Sony Bravia KD-50X85J_** running _**Android TV OS version 10**_.

- Disable _**Google Play Store**_.
- Open the **_smartTube_** app and run a backup: Settings --> Backup/Restore
- Uninstall _**smartTube**_.
- Install an older beta version of **_smartTube_**: eg. SmartTube_beta_21.45_armeabi-v7a.apk.
- Open **_smartTube_** and navigate to Settings --> About --> Check for updates.
- When prompted, set the option to allow **_startTube_** to have permissions to update.
- The update installed successfully: _**smartTube beta 26.80**_
- Enabled _**Google Play Store**_.

The only thing I did not do was to backup the smartTube setting before uninstalling the app.


For those that can not get the smartTube updating from the app, here is a linux bash script to do the job:
[smarttubeupdate.zip](https://github.com/user-attachments/files/19146721/smarttubeupdate.zip)

I have used this script to update smartTube from my ubuntu server.
You will need to install adb-tools (Android Debug Bridge) before running the script. Also, edit the script on line 8:
TVname=TV IP address or hostname here

Here is all the info one needs to install and run adb-tools:
https://developer.android.com/tools/adb

**Comment by melikootje** [comment]
> I know that it should be possible to disable play protect in Settings->Apps->Security&Restrictions on Android TV 10. But on my tv, there is only one entry in Security&Restrictions, which is "Unknown sources". There I can give single apps the permission to install apps from unknown sources. I had already allowed this for all apps from the start, but the play protect scan was made anyways and I could not install Smarttube.
> 
> However, I found a workaround: Under Settings->Apps->Google Play Store, I can "Force quit" and "Disable" the play store. I did this and installed Smarttube. No play protect scan was made, the installation was successful. Then I enabled the play store again, and Smarttube is still working.

I have currently the same issue as those above me except this fix which supposedly works for others wont work for me. It says 

![Image](https://github.com/user-attachments/assets/157e77db-0c9c-4aee-9473-d2083988cf41)

**Comment by melikootje** [comment]
adb gives this error

![Image](https://github.com/user-attachments/assets/e609b2ca-1563-4aa2-86c1-2758670e4dcd)

**Comment by dseven** [comment]
Above was not actually the same issue (error message is quite different) - was a case of trying to use the 64-bit APK on a 32-bit system.

**Comment by Strahinja** [comment]
I'm having a similar issue on EON SmartBox KSTB6020 (Android 9) with ST Beta since few weeks back. After attempted auto-update or "manual" installation from a file manager, I'm getting just a generic error message that the "App is not installed". I also tried to install older versions - 20.25 and 21.45 with the same results. I already have disabled the Google Play protection beforehand.

The only thing that worked was installing the Stable version of ST.
